apis/v1beta1/vspheremachinetemplate_webhook.go

```diff
@@ -68,6 +69,12 @@ func (v *VSphereMachineTemplateWebhook) ValidateCreate(_ context.Context, raw ru
 			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "template", "spec", "network", "devices", "ipAddrs"), "cannot be set in templates"))
 		}
 	}
+	if spec.VMXVersion != "" {
+		r := regexp.MustCompile("^vmx-[1-9][0-9]?$")
```

Comment:
we should drop the `?` as it'll allow `vmx-` as valid, which isn't

Reply:
Added unit tests to check this particular scenario, it is working fine. The `[1-9]` part makes sure that any digit (from 1 to 9) is needed and the question mark is for the second set `[0-9]` which is not necessary in case like `vmx-9` or `vmx-4`.